src/frontend/src/pages/FinancePage/CreateReimbursementRequestForm/CreateReimbursementRequestForm.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+import { useFieldArray, useForm } from 'react-hook-form';
+import { ClubAccount, ReimbursementProductCreateArgs, WbsNumber } from 'shared';
+import {
+  useCreateReimbursementRequest,
+  useGetAllExpenseTypes,
+  useGetAllVendors,
+  useUploadManyReceipts
+} from '../../../hooks/finance.hooks';
+import { useToast } from '../../../hooks/toasts.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import ErrorPage from '../../ErrorPage';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import CreateReimbursementRequestFormView from './CreateReimbursementFormView';
+import { useAllProjects } from '../../../hooks/projects.hooks';
+import { useHistory } from 'react-router-dom';
+import { routes } from '../../../utils/routes';
+
+export interface CreateReimbursementRequestFormInput {
+  vendorId: string;
+  account: ClubAccount;
+  dateOfExpense: Date;
+  expenseTypeId: string;
+  reimbursementProducts: ReimbursementProductCreateArgs[];
+  receiptFiles: {
+    file: File;
+  }[];
+}
+
+const schema = yup.object().shape({
+  vendorId: yup.string().required('Vendor is required'),
+  account: yup.string().required('Account is required'),
+  dateOfExpense: yup.date().required('Date of Expense is required'),
+  expenseTypeId: yup.string().required('Expense Type is required'),
+  reimbursementProducts: yup
+    .array()
+    .of(
+      yup.object().shape({
+        wbsNum: yup.object().required('WBS Number is required'),
+        name: yup.string().required('Description is required'),
+        cost: yup.number().required('Amount is required').min(1, 'Amount must be greater than 0')
+      })
+    )
+    .required('reimbursement products required')
+    .min(1, 'At least one Reimbursement Product is required'),
+  receiptFiles: yup.array().required('receipt files required').min(1, 'At least one Receipt is required')
+});
+
+const CreateReimbursementRequestForm = () => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      vendorId: '',
+      account: ClubAccount.CASH,
+      dateOfExpense: new Date(),
+      expenseTypeId: '',
+      reimbursementProducts: [] as ReimbursementProductCreateArgs[],
```

Comment:
should rename this type to just be ReimbursementProduct if we're using it in the frontend

Reply:
huh?

Reply:
I don't like using a variable with createArgs in it in the frontend, it's confusing

Reply:
Its a shared type so if i change it here id have to change it in the backend too